cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java

```diff
@@ -319,8 +319,9 @@ private void krNewBadGoalsError(Vertx vertx, VertxTestContext context, CruiseCon
                         } else {
                             context.failNow(new RuntimeException("This operation failed with an unexpected error:" + result.cause().getMessage()));
                         }
+                    } else {
```

Comment:
There's an even neater way to do this, since we have the test context you can have:

```
.onComplete(context.failing(throwable -> {
    if (throwable.getMessage().contains("java.lang.IllegalArgumentException: Missing hard goals")) {
        context.completeNow();
    } else {
        context.failNow(new RuntimeException("This operation failed with an unexpected error:" + throwable.getMessage()));
    });
```

Then if the call didn't fail the test vertx will fail the call automatically.

Reply:
Cool, pushed up the simplification. I also set the throwable as the cause of the new RuntimeException in case of null message.